openmls/src/group/mls_group/test_proposals.rs

```diff
@@ -312,3 +285,405 @@ fn proposal_queue_order() {
+    );
+    log::info!(" >>> Creating commit ...");
+    let params = CreateCommitParams::builder()
+        .framing_parameters(framing_parameters)
+        .credential_bundle(&alice_credential_bundle)
+        .proposal_store(&proposal_store)
+        .force_self_update(false)
+        .build();
+    let (mls_plaintext_commit, welcome_bundle_alice_bob_option, _kpb_option) = alice_group
+        .create_commit(params, &crypto)
+        .expect("Error creating commit");
+
+    log::info!(" >>> Staging & merging commit ...");
+
+    let staged_commit = alice_group
+        .stage_commit(&mls_plaintext_commit, &proposal_store, &[], None, &crypto)
+        .expect("error staging commit");
+    alice_group.merge_commit(staged_commit);
+    let ratchet_tree = alice_group.tree().public_key_tree_copy();
+
+    let _bob_group = MlsGroup::new_from_welcome(
+        welcome_bundle_alice_bob_option.unwrap(),
+        Some(ratchet_tree),
+        bob_key_package_bundle,
+        None,
+        &crypto,
+    )
+    .expect("Error joining group.");
+}
+
+#[test]
+fn test_group_context_extension_proposal_fails() {
+    let crypto = OpenMlsRustCrypto::default();
+    let ciphersuite = &Config::supported_ciphersuites()[0];
+
+    // Basic group setup.
+    let group_aad = b"Alice's test group";
+    let framing_parameters = FramingParameters::new(group_aad, WireFormat::MlsPlaintext);
+
+    let (alice_credential_bundle, alice_key_package_bundle) =
+        setup_client("Alice", ciphersuite, &crypto);
+    let (bob_credential_bundle, _) = setup_client("Bob", ciphersuite, &crypto);
+
+    let bob_key_package_bundle = KeyPackageBundle::new(
+        &[ciphersuite.name()],
+        &bob_credential_bundle,
+        &crypto,
+        vec![Extension::KeyPackageId(KeyIdExtension::default())],
+    )
+    .unwrap();
+    let bob_key_package = bob_key_package_bundle.key_package();
+
+    // Set required capabilities
+    let extensions = &[ExtensionType::Capabilities];
+    let proposals = &[
+        ProposalType::GroupContextExtensions,
+        ProposalType::Add,
+        ProposalType::Remove,
+        ProposalType::Update,
+    ];
+    let required_capabilities = RequiredCapabilitiesExtension::new(extensions, proposals);
+
+    let mut alice_group = MlsGroup::builder(GroupId::random(&crypto), alice_key_package_bundle)
+        .with_required_capabilities(required_capabilities)
+        .build(&crypto)
+        .expect("Error creating MlsGroup.");
+
+    // Alice tries to add a required capability it doesn't support itself.
+    let required_key_id = Extension::RequiredCapabilities(RequiredCapabilitiesExtension::new(
+        &[ExtensionType::KeyId],
+        &[],
+    ));
+    let e = alice_group.create_group_context_ext_proposal(
+        framing_parameters,
+        &alice_credential_bundle,
+        &[required_key_id.clone()],
+        &crypto,
+    ).expect_err("Alice was able to create a gce proposal with a required extensions it doesn't support.");
```

Comment:
```suggestion
    ).expect_err("Alice was able to create a gce proposal with a required extensions she doesn't support.");
```

Reply:
We should stop using alice and bob in the first place ...